File: src/main/java/com/shiminfxcvii/turing/enums/OrganizationBusinessStateEnum.java
package com.shiminfxcvii.turing.enums;

import lombok.Getter;

import java.util.Objects;

/**
 * 单位业务类型枚举类
 *
 * @author ShiminFXCVII
 * @since 1/4/2023 5:26 PM
 */
@Getter
public enum OrganizationBusinessStateEnum {

    AWAITING_CHECK("待审核"),
    PASSES("已通过"),
    RETURNED("已退回"),
    ;

    private static final OrganizationBusinessStateEnum[] VALUES = values();

    private final String desc;

    OrganizationBusinessStateEnum(String desc) {
        this.desc = desc;
    }

    public static String getValueByOrdinal(Integer ordinal) {
        // Use cached VALUES instead of values() to prevent array allocation.
        for (OrganizationBusinessStateEnum anEnum : VALUES)
            if (Objects.equals(anEnum.ordinal(), ordinal))
                return anEnum.getDesc();

        return "";
    }

}

File: src/main/java/com/simonvonxcvii/turing/enums/OrganizationBusinessStateEnum.kt
package com.simonvonxcvii.turing.enums

/**
 * 单位业务类型枚举类
 *
 * @author Simon Von
 * @since 1/4/2023 5:26 PM
 */
enum class OrganizationBusinessStateEnum(val value: String) {
    AWAITING_CHECK("待审核"),
    PASSES("已通过"),
    RETURNED("已退回"),

    /**
     * 未知
     */
    UNKNOWN("unknown"),
    ;

    companion object {
        private val VALUES = entries.toTypedArray()

        @JvmStatic
        fun getValueByOrdinal(ordinal: Int): String {
            // Use cached VALUES instead of values() to prevent array allocation.
            for (anEnum in VALUES)
                if (anEnum.ordinal == ordinal)
                    return anEnum.value
            return UNKNOWN.value
        }

        @JvmStatic
        fun getOrdinalByValue(value: String): Int {
            // Use cached VALUES instead of values() to prevent array allocation.
            for (anEnum in VALUES)
                if (anEnum.value == value)
                    return anEnum.ordinal
            return UNKNOWN.ordinal
        }

        @JvmStatic
        fun getEnumByOrdinal(ordinal: Int): OrganizationBusinessStateEnum {
            // Use cached VALUES instead of values() to prevent array allocation.
            for (anEnum in VALUES)
                if (anEnum.ordinal == ordinal)
                    return anEnum
            return UNKNOWN
        }

        @JvmStatic
        fun getEnumByValue(value: String): OrganizationBusinessStateEnum {
            // Use cached VALUES instead of values() to prevent array allocation.
            for (anEnum in VALUES)
                if (anEnum.value == value)
                    return anEnum
            return UNKNOWN
        }
    }
}
